pcap-linux.c

```diff
@@ -3419,10 +3431,16 @@ static int pcap_wait_for_frames_mmap(pcap_t *handle)
 	struct ifreq ifr;
 	int ret;
 	struct pollfd pollinfo[2];
+	int numpollinfo;
 	pollinfo[0].fd = handle->fd;
 	pollinfo[0].events = POLLIN;
-	pollinfo[1].fd = handlep->poll_breakloop_fd;
-	pollinfo[1].events = POLLIN;
+	if ( handlep->poll_breakloop_fd == -1 ) {
+		numpollinfo = 1;
```

Comment:
A colleague noticed that `pollinfo[1].revents` is tested below, so should be zeroed here.

Reply:
...with a comment noting that this is done so that both elements of `pollinfo[]` will be set after a successful call to `poll()` even when `numpollinfo` is 1.

(Setting them both requires extra stores but avoids a conditional branch, albeit one that, at least if the branch predictor can store enough predictions, will be predicted correctly most of the time; I'm not sure which is better from a performance point of view.

From a code-ugliness point of view, avoiding the `if` statement avoids another level of indirection.)

Reply:
I've added the code and the comment; please see if you think it's clear enough.

Reply:
Look good.